BackEnd/src/business/stocks.ts

```diff
@@ -0,0 +1,156 @@
+import TeamModel from '../../../Common/models/team.model';
+import { GetDbTeamRosterByTeamId, GetDbTeamsByTeamId } from '../db/teams';
+import { InvalidRequestError } from '../../../Common/errors';
+import { AuthUser } from './auth';
+import { buyDbStockForUser, createDBNewStockValue, getDbLatestStockValue, withdrawDbFunds } from '../db/stocks';
+import { GetRiotLeagueBySummonerId, GetRiotPlayerByPuuid } from '../external-api/player';
+import { RiotLeagueEntryDto } from '../../../Common/Interface/RiotAPI/RiotApiDto';
+import PlayerGameModel from '../../../Common/models/playergame.model';
+import { GetDbPlayerGamesByGameId } from '../db/games';
+import { ToGameId } from '../../../Common/utils';
+import { getDbPlayerTeamPlayerPuuid } from '../db/playerteam';
+import logger from '../../logger';
+
+export async function updateTeamStocksForGame(matchId: string): Promise<void> {
+  logger.info(`Updating Stock Values For Match: ${matchId}`);
+  const allPlayersGames: PlayerGameModel[] = await GetDbPlayerGamesByGameId(ToGameId(matchId));
```

Comment:
I think this can be done much easier by using the GameModel. It also has the season id and has a playersSummary. Which is a list of all players in the game by team.

Reply:
We need to use PlayerGameModel because we store what team was playing on at the time of match completion there. This information doesnt exist on the GameModel data. 

Since players can swap teams it isnt as easy as just querying the player_team DB.